From bb0be3664ebb5f55444246daf23b40997facf498 Mon Sep 17 00:00:00 2001
From: Domenico Acri <domenico.acri@engicam.com>
Date: Wed, 15 Mar 2017 17:55:32 +0100
Subject: [PATCH] fixed problem touchscreen goodix

---
 drivers/input/touchscreen/goodix.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index ef48bd4..40f0863 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -83,6 +83,10 @@ static int goodix_ts_read_input_report(struct goodix_ts_data *ts, u8 *data)
                 return error;        
 	}
 
+
+        if (!(data[0] & 0x80))
+                 return -EAGAIN;
+
         touch_num = data[0] & 0x0f;
         if (touch_num > GOODIX_MAX_CONTACTS)
                 return -EPROTO;
@@ -161,7 +165,8 @@ static irqreturn_t goodix_ts_irq_handler(int irq, void *dev_id)
 
        if (i2c_master_send(ts->client, end_cmd, sizeof(end_cmd)) < 0)
                dev_err(&ts->client->dev, "I2C write end_cmd error\n");
-
+	
+	msleep(2);
        return IRQ_HANDLED;
 }
 
@@ -340,7 +345,7 @@ static int goodix_ts_probe(struct i2c_client *client,
         if (error)
                 return error;
 
-        irq_flags = goodix_irq_flags[ts->int_trigger_type] | IRQF_ONESHOT;
+        irq_flags = IRQ_TYPE_EDGE_RISING | IRQF_ONESHOT;
         error = devm_request_threaded_irq(&ts->client->dev, client->irq,
                                           NULL, goodix_ts_irq_handler,
                                           irq_flags, client->name, ts);
-- 
2.7.4

